wallet/wallet_test.go

```diff
@@ -205,3 +206,89 @@ func TestLabelTransaction(t *testing.T) {
 		})
 	}
 }
+
+// TestGetTransaction tests if we can fetch a mined, an existing
+// and a non-existing transaction from the wallet like we expect.
+func TestGetTransaction(t *testing.T) {
+
+	tests := []struct {
+		name string
+
+		// Whether the transaction should be known to the wallet.
+		txKnown bool
+
+		// Whether the transaction is mined.
+		minedTx bool
+
+		// Block meta. Used to emulate mined transactions.
+		blockMeta *wtxmgr.BlockMeta
+
+		// The error we expect to be returned.
+		expectedErr error
+	}{
+		{
+			name:        "existing transaction, not mined",
+			txKnown:     true,
+			blockMeta:   nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "non-existing transaction",
+			txKnown:     false,
+			blockMeta:   nil,
+			expectedErr: ErrNoTx,
+		},
+		{
+			name:    "existing mined transaction",
+			txKnown: true,
+			blockMeta: &wtxmgr.BlockMeta{
+				Block: wtxmgr.Block{
+					Height: 100,
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			w, cleanup := testWallet(t)
+			defer cleanup()
+
+			// If the transaction should be known to the store, we
+			// write txdetail to disk.
+			if test.txKnown {
```

Comment:
this approach works, tho imo a better version would be creating testing a `txMined` and `txUnmined`, then define the table tests as 
```go
{
  name string
  txid chainhash.Hash
  expectedTx *GetTransactionResult
  expectedErr error
}
```
Then you construct three txids - `txMined = ...`, `txUnmined= ...`, and `txUnknown = ...`. And finally check return returned values from `GetTransaction`,
```go
require.ErrorIs(t, tc.expectedErr, err)
require.Equal(t, tc.expectedTx, tx)
```

Reply:
@yyforyongyu I have updated the test code. I am not sure if you wanted me to add all transactions to the wallet storage  at once or one for each test cycle like I have done now?

Reply:
I had issues storing the full GetTransactionResults array and opted for comparing the txid and block height as validators.